code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -572,4 +585,25 @@
 	..()
 	if(AIStatus == AI_Z_OFF)
 		SSidlenpcpool.idle_mobs_by_zlevel[old_z] -= src
-		toggle_ai(initial(AIStatus))
\ No newline at end of file
+		toggle_ai(initial(AIStatus))
+
+/mob/living/simple_animal/proc/add_collar(obj/item/clothing/accessory/petcollar/P, mob/user)
+	if(QDELETED(P) || pcollar)
+		return
+	if(!user.unEquip(P))

```

Comment:
should probably have a user check here for if we want to use this proc for non-users.